### PR TITLE
categorised loot tables for mapping

### PR DESCRIPTION
**Pull request by Stealthkibbler**
pring it so floyd can pick it apart and say what needs to be tweaked changed

### CODE REVIEW

**Comment by ninjanomnom** [comment]
if you're redoing loot drops you may as well make the amounts random in the new proc

**Comment by Stealthkibbler** [comment]
no u

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 1]
```diff
@@ -296,3 +296,411 @@
...
+			/obj/item/stack/sheet/mineral/gold{amount = 5} = 5,
+			/obj/item/stack/sheet/mineral/silver{amount = 5} = 5,
+			/obj/item/stack/sheet/mineral/diamond{amount = 1} = 3,
+			/obj/item/stack/sheet/mineral/plasma{amount = 10} = 4,
+			/obj/item/stack/sheet/metal{amount = 25} = 7,
+			/obj/item/stack/sheet/plasteel{amount = 10} = 5,
+			/obj/item/stack/sheet/glass{amount = 35} = 7,
+			/obj/item/stack/sheet/rglass{amount = 10} = 5,
+			/obj/item/stack/sheet/mineral/wood{amount = 50} = 7,
+			/obj/item/stack/sheet/mineral/abductor{amount = 5} = 1,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: that's fucking 50k, can you not?

delete this its dumb.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 2]
```diff
@@ -296,3 +296,411 @@
...
+			/obj/item/stack/sheet/mineral/mythril{amount = 1} = 1,
+			/obj/item/stack/sheet/mineral/gold{amount = 5} = 5,
+			/obj/item/stack/sheet/mineral/silver{amount = 5} = 5,
+			/obj/item/stack/sheet/mineral/diamond{amount = 1} = 3,
+			/obj/item/stack/sheet/mineral/plasma{amount = 10} = 4,
+			/obj/item/stack/sheet/metal{amount = 25} = 7,
+			/obj/item/stack/sheet/plasteel{amount = 10} = 5,
+			/obj/item/stack/sheet/glass{amount = 35} = 7,
+			/obj/item/stack/sheet/rglass{amount = 10} = 5,
+			/obj/item/stack/sheet/mineral/wood{amount = 50} = 7,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: too large amounts for wood metal and glass

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 3]
```diff
@@ -296,3 +296,411 @@
...
+			/obj/item/weapon/coin/plasma = 3,
+			/obj/item/weapon/coin/diamond = 3,
+			/obj/item/weapon/coin/mythril = 2,
+			/obj/item/weapon/coin/adamantine = 2,
+			/obj/item/weapon/coin/clown = 1,
+			)
+
+
+/obj/effect/spawner/lootdrop/illegaltech_loot_spawner // Contains contraband technology manafactured by the Syndicate, not always necessarily dangerous, but sometimes extremely valuable for R&D
+		name = "illegaltech salvage spawner"
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: fix this

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 4]
```diff
@@ -296,3 +296,411 @@
...
+/obj/effect/spawner/lootdrop/abductor // Includes all abductor materials and equipment
+		name ="abductor salvage spawner"
+		lootdoubles = 0
+		loot = list(
+			/obj/item/clothing/suit/armor/abductor/vest = 3,
+			/obj/item/device/abductor/gizmo = 1,
+			/obj/item/device/abductor/silencer = 1,
+			/obj/item/device/firing_pin/abductor = 6,
+			/obj/item/organ/tongue/abductor = 1,
+			/obj/item/stack/sheet/mineral/abductor{amount = 25} = 2,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: are you mentally deficit? THATS 250K IN FUNDS. D E L E T E T H I S

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 5]
```diff
@@ -296,3 +296,411 @@
...
+			/obj/item/organ/appendix = 1,
+			)
+
+/obj/effect/spawner/lootdrop/mechs // lootdrop for mechs
+		name ="mech salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/mecha/combat/durand = 2,
+			/obj/mecha/combat/gygax = 2,
+			/obj/mecha/combat/gygax/dark = 1,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: this needs a huge ass empty slot with a high weight. mechs should be rare.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 6]
```diff
@@ -296,3 +296,411 @@
...
+			)
+
+/obj/effect/spawner/lootdrop/department/science // tech, advanced stock parts and other knick knacks
+		name ="science department salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/item/weapon/stock_parts/manipulator/femto = 3,
+			/obj/item/weapon/stock_parts/matter_bin/bluespace = 3,
+			/obj/item/weapon/stock_parts/micro_laser/quadultra = 3,
+			/obj/item/weapon/stock_parts/scanning_module/triphasic = 3,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: needs huge ass empty weights, atleast 25 in empty weight,

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 7]
```diff
@@ -296,3 +296,411 @@
...
+			/obj/structure/closet/crate/medical = 4,
+			/obj/structure/closet/crate/internals = 3,
+			)
+
+
+/obj/effect/spawner/lootdrop/shells // munition shells
+		name ="munition shells salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/structure/shell = 1,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: weird distribution, just make it equal.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 8]
```diff
@@ -296,3 +296,411 @@
...
+/obj/effect/spawner/lootdrop/slime_extracts // more controlled way of spawning slime extracts
+		name ="slime extracts salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/item/slime_extract/adamantine = 2,
+			/obj/item/slime_extract/black = 3,
+			/obj/item/slime_extract/blue = 4,
+			/obj/item/slime_extract/bluespace = 3,
+			/obj/item/slime_extract/cerulean = 4,
+			/obj/item/slime_extract/darkblue = 5,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: needs empty weight

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 9]
```diff
@@ -296,3 +296,411 @@
...
+		name ="slime potions salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/item/clothing/gloves/color/rainbow/clown = 3,
+			/obj/item/clothing/shoes/clown_shoes/banana_shoes = 2,
+			/obj/item/device/firing_pin/clown = 3,
+			/obj/item/weapon/reagent_containers/food/snacks/grown/banana/bluespace = 2,
+			/obj/item/mine/spawner/banana = 2,
+			/obj/item/stack/sheet/mineral/bananium{amount = 25}= 1,
+			/obj/item/weapon/grenade/clusterbuster/soap = 1,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: empty weight needed and name is wrong.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 10]
```diff
@@ -296,3 +296,411 @@
...
+/obj/effect/spawner/lootdrop/weapons/guns_light // sidearms that have rather limited ammunition and usually can't finish the job
+		name ="light gun salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/item/weapon/gun/projectile/automatic/pistol = 2,
+			/obj/item/weapon/gun/projectile/automatic/pistol/APS = 1,
+			/obj/item/weapon/gun/projectile/automatic/pistol/m1911 = 3,
+			/obj/item/weapon/gun/projectile/automatic/pistol/c05r = 1,
+			/obj/item/weapon/gun/energy/gun/mini = 2,
+			/obj/item/weapon/gun/energy/gun/advtaser = 3,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: empty weight needed.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 11]
```diff
@@ -296,3 +296,411 @@
...
+		loot = list(
+			/obj/item/weapon/gun/projectile/automatic/proto/unrestricted = 1,
+			/obj/item/weapon/gun/projectile/automatic/c20r/unrestricted = 1,
+			/obj/item/weapon/gun/projectile/revolver = 2,
+			/obj/item/weapon/gun/projectile/automatic/wt550 = 3,
+			/obj/item/weapon/gun/projectile/automatic/shotgun/bulldog/unrestricted = 2,
+			/obj/item/weapon/gun/projectile/shotgun/lethal = 3,
+			/obj/item/weapon/gun/energy/gun = 2,
+			/obj/item/weapon/gun/energy/laser = 2,
+			)
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: empty weight needed.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 12]
```diff
@@ -296,3 +296,411 @@
...
+
+/obj/effect/spawner/lootdrop/weapons/guns_deadly // designed to be the most effective at killing, with balance in mind
+		name ="deadly gun salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/item/weapon/gun/projectile/automatic/m90/unrestricted = 1,
+			/obj/item/weapon/gun/projectile/shotgun/automatic/combat = 2,
+			/obj/item/weapon/gun/projectile/revolver/mateba = 3,
+			/obj/item/weapon/gun/projectile/automatic/sniper_rifle = 1,
+			/obj/item/weapon/gun/projectile/automatic/l6_saw/unrestricted = 1,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: EMPTY WEIGHT HOLY FUCK, NEEDS ATLEAST 50 EMPTY WEIGHT

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 13]
```diff
@@ -296,3 +296,411 @@
...
+			)
+
+/obj/effect/spawner/lootdrop/weapons/guns_ludicrous // guns that are designed to be unfair and have little to none balance
+		name ="ludicrous gun salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/item/weapon/gun/projectile/automatic/xmg80 = 3,
+			/obj/item/weapon/gun/projectile/automatic/aks74 = 2,
+			/obj/item/weapon/gun/projectile/automatic/ar = 2,
+			/obj/item/weapon/gun/energy/pulse/carbine = 1,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: 100 empty weight thanks.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 14]
```diff
@@ -296,3 +296,411 @@
...
+			)
+
+/obj/effect/spawner/lootdrop/weapons/melee_weapons // not enough variety with melee to split it into seperate categories
+		name ="melee weapons salvage spawner"
+		lootdoubles = 1
+		loot = list(
+			/obj/item/weapon/melee/energy/sword/saber = 1,
+			/obj/item/weapon/melee/rapier = 3,
+			/obj/item/weapon/katana = 2,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: 10 empty weight

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 15]
```diff
@@ -296,3 +296,411 @@
...
+			/obj/item/stack/spacecash/c200 = 5,
+			/obj/item/stack/spacecash/c500 = 4,
+			/obj/item/stack/spacecash/c1000 = 2,
+			/obj/item/weapon/coin/iron = 7,
+			/obj/item/weapon/coin/silver = 7,
+			/obj/item/weapon/coin/uranium = 7,
+			/obj/item/weapon/coin/gold = 5,
+			/obj/item/weapon/coin/plasma = 3,
+			/obj/item/weapon/coin/diamond = 3,
+			/obj/item/weapon/coin/mythril = 2,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: needs some empty weight

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 16]
```diff
@@ -296,3 +296,411 @@
...
+			/obj/item/weapon/soap/syndie = 7,
+			/obj/item/toy/carpplushie/dehy_carp = 7,
+			/obj/item/device/flashlight/emp = 5,
+			/obj/item/weapon/storage/box/syndie_kit/cutouts = 7,
+			/obj/item/weapon/storage/box/syndie_kit/space = 6,
+			/obj/item/clothing/suit/space/hardsuit/syndi = 2,
+			/obj/item/clothing/suit/space/hardsuit/syndi/elite = 1,
+			/obj/item/weapon/storage/belt/military = 6,
+			/obj/item/device/encryptionkey/binary = 5,
+			/obj/item/device/multitool/ai_detect = 6,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: needs empty weight

File: code/game/objects/effects/spawners/lootdrop.dm (hunk 4 above)
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: also this list in general is a meme

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 17]
```diff
@@ -296,3 +296,411 @@
...
+			/obj/item/weapon/storage/firstaid/toxin = 4,
+			/obj/item/stack/medical/bruise_pack = 5,
+			/obj/item/stack/medical/ointment = 5,
+			/obj/item/stack/medical/gauze = 5,
+			/obj/item/weapon/reagent_containers/syringe/bluespace = 1,
+			/obj/item/weapon/storage/pill_bottle/stimulant = 1,
+			/obj/item/weapon/reagent_containers/glass/beaker/bluespace = 1,
+			/obj/item/weapon/reagent_containers/hypospray/combat/nanites = 1,
+			/obj/item/weapon/defibrillator/compact/combat/loaded = 3,
+			/obj/item/weapon/gun/medbeam = 2,
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: no medbeam. remove this.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 18]
```diff
@@ -296,3 +296,411 @@
...
+		lootdoubles = 1
+		loot = list(
+			/obj/structure/closet/crate/critter = 1 ,
+			/obj/structure/closet/crate/large = 5,
+			/obj/structure/closet/crate/secure/plasma = 1,
+			/obj/structure/closet/crate/secure/loot = 3,
+			/obj/structure/closet/crate/secure/weapon = 1,
+			/obj/structure/closet/crate/medical = 4,
+			/obj/structure/closet/crate/internals = 3,
+			)
```
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: -facepalm- those are empty you fucking doofus

File: code/game/objects/effects/spawners/lootdrop.dm (hunk 13 above)
Comment (Qustinnus) [inline code/game/objects/effects/spawners/lootdrop.dm]: also no fucking pulse rifles

File: code/game/objects/effects/spawners/lootdrop.dm (hunk 5 above)
Comment (ninjanomnom) [inline code/game/objects/effects/spawners/lootdrop.dm]: mech wreckage would be good too

**Pull request closed without merging**